ambry-rest/src/main/java/com.github.ambry.rest/StorageServerNettyFactory.java

```diff
@@ -0,0 +1,88 @@
+/**
+ * Copyright 2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.rest;
+
+import com.codahale.metrics.MetricRegistry;
+import com.github.ambry.commons.SSLFactory;
+import com.github.ambry.config.NettyConfig;
+import com.github.ambry.config.PerformanceConfig;
+import com.github.ambry.config.VerifiableProperties;
+import io.netty.channel.ChannelInitializer;
+import io.netty.channel.socket.SocketChannel;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * Netty specific implementation of {@link NioServerFactory}.
+ * <p/>
+ * Sets up all the supporting cast required for the operation of {@link NettyServer} and returns a new instance on
+ * {@link #getNioServer()}.
+ */
+public class StorageServerNettyFactory implements NioServerFactory {
+  private static final Logger LOGGER = LoggerFactory.getLogger(StorageServerNettyFactory.class);
+
+  private final NettyConfig nettyConfig;
+  private final PerformanceConfig performanceConfig;
+  private final NettyMetrics nettyMetrics;
+  final Map<Integer, ChannelInitializer<SocketChannel>> channelInitializers;
+
+  /**
+   * Creates a new instance of NettyFrontendServerFactory.
+   * @param http2Port the port for HTTP2 request.
+   * @param verifiableProperties the in-memory {@link VerifiableProperties} to use.
+   * @param metricRegistry the {@link MetricRegistry} to use.
+   * @param requestHandler the {@link RestRequestHandler} to hand off the requests to.
+   * @param publicAccessLogger the {@link PublicAccessLogger} that can be used for public access logging
+   * @param restServerState the {@link RestServerState} that can be used to check the health of the system
+   *                              to respond to health check requests
+   * @param defaultSslFactory the {@link SSLFactory} used to construct the {@link javax.net.ssl.SSLEngine} used for
```

Comment:
It does not look like this factory has the override behavior, so you can modify this javadoc and change the arg name to just `sslFactory`